debugger:native is currently unusable, coupled with gc and types diagnostics

`cpp` backend
just adding a `--debugger:native` will result in `multiple definition of nimTypeRoot`

Really we need a test for those critical things. Cpp should be more tested before any serious release.

I suspect that the fix or work related to #8964 messed up things a bit.
adding `importc` to the `nimTypeRoot` var to make it extern in every compiled modules var and emitting a global `nimTypeRoot` somewhere fixes it.

--- a/lib/system/hti.nim
+++ b/lib/system/hti.nim
@@ -103,6 +103,10 @@ type
   PNimType = ptr TNimType
 
 when defined(nimTypeNames):
-  var nimTypeRoot {.compilerProc.}: PNimType
+  # Declare this variable only once in system.nim
+  when declared(ThisIsSystem):
+    var nimTypeRoot {.compilerProc.}: PNimType
+  else:
+    var nimTypeRoot {.importc.}: PNimType
 
 # node.len may be the ``first`` element of a set

--- a/tests/cpp/t9013.nim
+++ b/tests/cpp/t9013.nim
@@ -1,0 +1,9 @@
+discard """
+  targets: "cpp"
+  cmd: "nim $target --debugger:native $options $file"
+"""
+
+# The --debugger switch is needed in order to enable the defined(nimTypeNames)
+# code path in hti.nim
+import typeinfo
+var tt: Any

